gapic-generator-java/src/main/java/com/google/api/generator/gapic/protoparser/Parser.java

```diff
@@ -683,9 +689,28 @@ static List<Method> parseMethods(
       Map<String, Message> messageTypes,
       Map<String, ResourceName> resourceNames,
       Optional<GapicServiceConfig> serviceConfigOpt,
+      Optional<com.google.api.Service> serviceYamlProtoOpt,
       Set<ResourceName> outputArgResourceNames,
       Transport transport) {
     List<Method> methods = new ArrayList<>();
+
+    // Parse the serviceYaml for autopopulated methods and fields once and put into a map
+    ImmutableMap.Builder<String, List<String>> autoPopulatedMethodsWithFieldsBuilder =
+        ImmutableMap.builder();
+    if (serviceYamlProtoOpt.isPresent()) {
```

Comment:
There are three nested indentations, can we extract this logic to a separate method so it's more maintainable and testable?